src/Apps/Consign/Routes/SubmissionFlow/UploadPhotos/UploadPhotos.tsx

```diff
@@ -384,6 +392,32 @@ export const UploadPhotos: React.FC<UploadPhotosProps> = ({
                 disabled={isSubmitting || !isValid}
                 loading={isSubmitting || values.photos.some(c => c.loading)}
                 type="submit"
+                onClick={event => {
+                  if (!isLoggedIn) {
+                    event.preventDefault()
+
+                    showAuthDialog({
+                      mode: "Login",
+                      options: {
+                        title: mode =>
+                          mode === "Login"
+                            ? "Log in to submit for sale"
+                            : "Sign up to submit for sale",
+                          afterAuthAction: {
+                            action: "associateSubmission",
+                            kind: "submission",
+                            objectId: submission?.externalId!,
```

Comment:
Note that while this "associateSubmission" afterAuthAction is already defined, it doesn't seem to be used and currently assumes that we've already collected the visitor's email address in the flow. We recently removed the Contact Information step in the flow so we no longer collect the email address in the UI.

This change will need to be paired with a corresponding PR in Convection which currently defines logic to allow visitor submissions to be later associated with an Artsy account. I'll provide the link to that PR here when ready.

Reply:
PR for that Convection change [is here](https://github.com/artsy/convection/pull/1480).

Reply:
thanks for the comment

Reply:
can we also navigate the user to `Thank your submitting` step after the sign in/up and we associate the submission to their user.

Reply:
It's a bit tricky to add more after auth behavior here. We'd have to 1) associate the submission 2) submit the submission and 3) navigate to next step, all in the after auth callback.

As this will be soon be replaced by the redesigned flow, let's add any new behavior to that instead of adding to the behavior here.